cmd/apiserver/app/server/util.go

```diff
@@ -17,48 +17,154 @@ limitations under the License.
+// serviceCatalogConfig is a placeholder for configuration
+type serviceCatalogConfig struct {
+	// the shared informers that know how to speak back to this apiserver
+	sharedInformers informers.SharedInformerFactory
+	// the shared informers that know how to speak back to kube apiserver
+	kubeSharedInformers kubeinformers.SharedInformerFactory
+	// the configured loopback client for this apiserver
+	client internalclientset.Interface
+	// the configured client for kube apiserver
+	kubeClient kubeclientset.Interface
+}
+
+// standaloneMode returns true if the env var SERVICE_CATALOG_STANDALONE
+// is present.  if present, we will assume no integration with kubernetes
+// api server is available.  this is primarily useful for testing purposes.
+func standaloneMode() bool {
+	val := os.Getenv("SERVICE_CATALOG_STANDALONE")
+	return len(val) > 0
+}
+
+// buildGenericConfig takes the server options and produces the genericapiserver.Config associated with it
+func buildGenericConfig(s *ServiceCatalogServerOptions) (*genericapiserver.Config, *serviceCatalogConfig, error) {
+	// check if we are running in standalone mode (for test scenarios)
+	inCluster := !s.StandaloneMode
+	if !inCluster {
+		glog.Infof("service catalog is in standalone mode")
+	}
 	if _, err := s.SecureServingOptions.ServingOptions.DefaultExternalAddress(); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-
 	// server configuration options
-	if err := s.SecureServingOptions.MaybeDefaultWithSelfSignedCerts(
-		s.GenericServerRunOptions.AdvertiseAddress.String(),
-	); err != nil {
-		return nil, err
+	if err := s.SecureServingOptions.MaybeDefaultWithSelfSignedCerts(s.GenericServerRunOptions.AdvertiseAddress.String()); err != nil {
+		return nil, nil, err
 	}
+	// NOTE: in k8s 1.7, this will take the explicit codec as input
 	genericConfig := genericapiserver.NewConfig()
 	if err := s.GenericServerRunOptions.ApplyTo(genericConfig); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	if err := s.SecureServingOptions.ApplyTo(genericConfig); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-
+	// this MUST be done after secure so we have a valid loopbackClientConfig
 	if err := s.InsecureServingOptions.ApplyTo(genericConfig); err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+	if inCluster {
+		if err := s.AuthenticationOptions.ApplyTo(genericConfig); err != nil {
+			return nil, nil, err
+		}
+	}
+	// TODO: add support for audit log options
+	// TODO: add support for OpenAPI config
+	genericConfig.SwaggerConfig = genericapiserver.DefaultSwaggerConfig()
+	// TODO: investigate if we need metrics unique to service catalog, but take defaults for now
+	genericConfig.EnableMetrics = true
+	// TODO: add support to default these values in build
+	serviceCatalogVersion := version.Get()
+	genericConfig.Version = &serviceCatalogVersion
+
+	// FUTURE: use protobuf for communication back to itself?
```

Comment:
i was less concerned about it in near term until we do a lot of loopback communication.